sabnzbd/filesystem.py

```diff
@@ -166,7 +166,25 @@ def sanitize_filename(name: str) -> str:
     if not name:
         name = "unknown"
 
+    # now split name into name, ext
     name, ext = os.path.splitext(name)
+
+    # If filename is too long, brute-force truncate it:
+    if len(name) + len(ext) > DEF_FILE_MAX:
+        logging.debug("Filename %s is too long, so truncating", name + ext)
+        # Too long filenames are often caused by incorrect non-ascii chars,
+        # so brute-force remove those non-ascii chars
+        name = str(name.encode("ascii", "ignore"), "utf-8")
+        if len(name) + len(ext) > DEF_FILE_MAX:
+            # still too long, limit the extension
+            maxextlength = 100  # max length of an extension
```

Comment:
Maybe go with a more realistic limit here? After all, we only really care about the extension because it may provide useful information; in all other cases it's just a random dot somewhere in the filename.

Reply:
OK. I checked my files, and the longest extension I had was "releaseinfo". 

So ... limit to ... 20?

Reply:
If anything, that would still leave plenty of room. But definitely better than 100.